### PR TITLE
Fix pubsub and update dependency

### LINKED ISSUE #1939: java error with pubsub connector in 1.39
We got issue after upgrading from 1.38.0 to 1.39.2

```
2022-11-03 19:37:04 INFO  Maxwell - Starting Maxwell. maxMemory: 518979584 bufferMemoryUsage: 0.25
2022-11-03 19:37:07 INFO  TaskManager - Stopping 0 tasks
2022-11-03 19:37:07 INFO  TaskManager - Stopped all tasks
Exception in thread "main" java.util.ServiceConfigurationError: io.grpc.NameResolverProvider: io.grpc.grpclb.SecretGrpclbNameResolverProvider$Provider Unable to get public no-arg constructor
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
	at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:673)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1233)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
	at io.grpc.ServiceProviders.loadAll(ServiceProviders.java:67)
	at io.grpc.NameResolverRegistry.getDefaultRegistry(NameResolverRegistry.java:101)
	at io.grpc.internal.AbstractManagedChannelImplBuilder.<init>(AbstractManagedChannelImplBuilder.java:114)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.<init>(NettyChannelBuilder.java:136)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.<init>(NettyChannelBuilder.java:131)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.forAddress(NettyChannelBuilder.java:117)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider.builderForAddress(NettyChannelProvider.java:37)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider.builderForAddress(NettyChannelProvider.java:23)
	at io.grpc.ManagedChannelBuilder.forAddress(ManagedChannelBuilder.java:39)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createSingleChannel(InstantiatingGrpcChannelProvider.java:350)
	at com.google.api.gax.grpc.ChannelPool.<init>(ChannelPool.java:105)
	at com.google.api.gax.grpc.ChannelPool.create(ChannelPool.java:83)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createChannel(InstantiatingGrpcChannelProvider.java:236)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.getTransportChannel(InstantiatingGrpcChannelProvider.java:230)
	at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:201)
	at com.google.cloud.pubsub.v1.stub.GrpcPublisherStub.create(GrpcPublisherStub.java:203)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:189)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:89)
	at com.google.cloud.pubsub.v1.Publisher$Builder.build(Publisher.java:838)
	at com.zendesk.maxwell.producer.MaxwellPubsubProducerWorker.<init>(MaxwellPubsubProducer.java:192)
	at com.zendesk.maxwell.producer.MaxwellPubsubProducer.<init>(MaxwellPubsubProducer.java:111)
	at com.zendesk.maxwell.MaxwellContext.getProducer(MaxwellContext.java:538)
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:257)
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:227)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:337)
Caused by: java.lang.VerifyError: Cannot inherit from final class
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getConstructor(Class.java:2151)
	at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:660)
	at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:657)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:668)
	... 29 more
```

We're using maxwell to read from MysQL and to push to GCP pubsub.
Our maxwell process is hosted on k8s and using docker.io official images.

### PR DESCRIPTION
**Pull request by mozarik**
This fix #1939 and update to #1940. 

I change `pubsubRetryDelayMultiplier` default value to 1.3f to follow default on PubSub lib

This also fix failed to auth to google when deployed to kubernetes. 

`$GOOGLE_APPLICATION_CREDENTIALS file is in "volumeMount" `

### CODE REVIEW

**Comment by mozarik** [comment]
@osheroff can please review this fix

File: pom.xml [hunk 1]
```diff
@@ -343,14 +358,67 @@
...
+      <groupId>org.conscrypt</groupId>
+      <artifactId>conscrypt-openjdk-uber</artifactId>
+      <version>2.5.2</version>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.checkerframework</groupId>
+      <artifactId>checker-qual</artifactId>
+      <version>3.27.0</version>
+    </dependency>
```
Comment (osheroff) [inline pom.xml:415]: huh.  what's this?

File: pom.xml [hunk 2]
```diff
@@ -343,14 +358,67 @@
...
+      <groupId>io.grpc</groupId>
+      <artifactId>grpc-context</artifactId>
+      <version>1.50.2</version>
+    </dependency>
+    <dependency>
+      <groupId>org.conscrypt</groupId>
+      <artifactId>conscrypt-openjdk-uber</artifactId>
+      <version>2.5.2</version>
+      <scope>runtime</scope>
+    </dependency>
```
Comment (osheroff) [inline pom.xml:410]: and this?

File: pom.xml [hunk 3]
```diff
@@ -343,14 +358,67 @@
...
+    <dependency>
+      <groupId>org.checkerframework</groupId>
+      <artifactId>checker-qual</artifactId>
+      <version>3.27.0</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.errorprone</groupId>
+      <artifactId>error_prone_annotations</artifactId>
+      <version>2.16</version>
+    </dependency>
```
Comment (osheroff) [inline pom.xml:420]: ?

**Pull request closed without merging**